### ISSUE TITLE
Missing `aria-invalid` and `aria-required` on some XBL controls

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#5929]
The `aria-invalid` and `aria-required` should go on the field for components with just one field, on the element with `role="group"` for components with multiple fields¹.

This should be handled by the XBL engine. On the `<xbl:binding>`, the XBL engine support a `xxbl:label-for` pointing to an element, on which the XBL engine puts a `aria-labelledby` pointing to the label for the control. I _think_ that the `aria-invalid` and `aria-required` should go on that same element. If so, instead of adding an `xxbl:aria-invalid-elem` and `xxbl:aria-required-elem`, could we just have a, say, `xxbl:aria-elem`, that takes care of the `aria-labelledby`, `aria-invalid`, and `aria-required`?

Notes:

1. We shouldn't put the `aria-invalid` on the subfields, as for now we don't know which subfield is invalid. We could also put the `aria-required` on the individual fields, but I don't think it is necessary.

**Comment by ebruchez** [comment]
- [ ] start for example with `fr:number`

**Comment by ebruchez** [comment]
In the case of `fr:number`, we have `xxbl:label-for="html-input"`. Here, clearly, the `aria-*` attributes must end up on the HTML `input` element.

We could

- use AVTs to show/hide those attributes
- OR we could do this natively at the XBL level (which is probably better)

Two places where to handle this:

- page load: output attributes on the HTML element
    - we are `xxf:control="true"` here so we must have an element
- page update: update attributes

**Comment by ebruchez** [comment]
Now internally, a `xxf:control="true"` element is handled as an `xf:group`, and `handleAriaAttributes()`, which handles `aria-required` and `aria-invalid`, is not called for groups. It is called for:

- `xf:input`
- `xf:secret`
- `xf:select`/`xf:select1`
- `xf:textarea`
- Q: Should it be used for `xf:output`?

We use `xxf:control="true"` so we can have automatic handling of `aria-labeledby` and `aria-describedby` (I think). But we don't really intend or need for this to behave like a "group".

Can we call `handleAriaAttributes()` in any case? Except we might not want all `xxf:control="true"` elements to have these CSS classes. This would require also checking why we use `xxf:control="true"` in general.

**Comment by ebruchez** [comment]
Checking usages, it seems we mostly have cases where we just want to control an `xf:group`'s element name. Exceptions are:

- `fr:date`
- `fr:time`
- `fr:number`

**Comment by avernet** [comment]
[+1 from customer](https://3.basecamp.com/3600924/buckets/1966718/messages/6338049163)

**Comment by ebruchez** [comment]
Duplicate of #6279.

**Comment by ebruchez** [comment]
Checked that this works for:

- `fr:date`
- `fr:time`
- `fr:number`

**Comment by ebruchez** [comment]
Closing as duplicate.
